### DISCUSSION TITLE
Should I migrate from Vue 2 to Vue 3 or React?

### DISCUSSION THREAD

**Discussion by maaaaaars** [community/community#109371] [Programming Help]
Our team took over a Laravel project with the frontend built with Vue 2 and Quasar. All developers on the team can continue developing in Vue but don't like Quasar's components and want to migrate the project to Vue 3.

However, the overall team preference is React over Vue for several reasons:
- We will be able to grow the team more easily (larger pool of developers), allowing us to improve the project more efficiently in the long run.
- Every developer on the team is more comfortable with React.
- We believe we'll have more and better components available.
- About half of the team thinks React will offer better performance and a more polished look.
- About half of the team thinks migrating to React would take the same amount of time as migrating to Vue 3.

We have a loose deadline but would like to spend as little time as possible on migration. Furthermore, the project's performance is the most critical aspect.

We know the challenges we must face with migration to either React or Vue 3 and are looking for the following information:
1. Which will be quicker, migrating to React or Vue 3, or would it really take the same amount of time?
2. Which will offer better performance as the project scales to support thousands, potentially millions, of users?
3. Which will make it easier to grow the team, stay on track with deadlines, and resolve bugs efficiently?

I know there are no right or wrong answers here, so please share your opinions, no matter how subjective. I also don't know if this is the right place to ask this question. If you know a better place, let me know.

**Comment by spenserblack** [comment]
I can't speak for transitioning from Vue 2 to React, but I found transitioning from Vue 2 to Vue 3 to be a fairly easy process. IIRC most of the Vue 2 usage could migrate with some small changes, and then you can choose to move to the features and simplified usage introduced in Vue 3 at a later date. However, since you *don't* like Quasar's components, it sounds like you might want to drop it, which means you'll probably be doing a major rewrite anyway. So ease of migration might not apply so much.

**Reply by maaaaaars** [reply]
Thanks for your reply. I'm in favor of making our own forms, inputs, cards, etc. to gain full control. It would take time, but I think it would also pay off in the long run.

If we (hypothetically) drop Quasar, do you think it would make sense to migrate from Vue to React?

**Reply by spenserblack** [reply]
I've barely used React, so I'm afraid it will be pretty much impossible to answer that without some bias :sweat_smile:

But I will say that I have enjoyed Vue 3 a lot when writing SFCs (Single-File Components). It just feels a bit easier to organize IMO. For example:

```vue
<script setup>
import { computed, ref } from 'vue'

const name = ref('World');
const greeting = computed(() => `Hello, ${name.value}!`);
const greet = () => window.alert(greeting.value);
</script>

<template>
  <h1>{{ greeting }}</h1>
  <label>Name: <input v-model="name" /></label>
  <button @click="greet">Get greeting</button>
</template>

<style scoped>
  h1 {
    font-style: italic;
  }
</style>
```

But this is just a personal preference of mine, and shouldn't outweigh the various concerns that you've listed.

**Reply by maaaaaars** [reply]
Hah, fair enough. I've never used Vue, but I'm willing to learn if we decide to stick with it. Thanks for sharing your opinion . I appreciate it.
